nilearn/glm/thresholding.py

```diff
@@ -186,8 +167,14 @@ def cluster_level_inference(
     .. footbibliography::
 
     """
-    parameters = dict(**inspect.signature(cluster_level_inference).parameters)
-    warn_default_threshold(threshold, parameters["threshold"].default, 3.0)
+    # TODO (nilearn >= 0.15.0) remove
+    warnings.warn(
+        "From nilearn version>=0.15, "
+        "the default 'threshold' will be set to "
+        f"{DEFAULT_Z_THRESHOLD}.",
+        FutureWarning,
+        stacklevel=find_stack_level(),
+    )
```

Comment:
Changed the code to raise the warning whatever value specified for `threshold`.

Comparing with default value causes conflicts.